go-controller/pkg/ovn/egressgw.go

```diff
@@ -310,7 +324,14 @@ func (oc *DefaultNetworkController) deletePodGWRoute(routeInfo *apbroutecontroll
 	if utilnet.IsIPv6String(gw) != utilnet.IsIPv6String(podIP) {
 		return nil
 	}
-
+	local, err := oc.isPodInLocalZone(routeInfo.PodName)
```

Comment:
here in the delete path as well, if the error is api not found, we need to continue to try to delete